src/homepageExperience/components/steps/cli/InitializeClient.tsx

```diff
@@ -67,6 +80,15 @@ export const InitializeClient: FC<OwnProps> = ({
     () => allPermissionTypes.sort((a, b) => collator.compare(a, b)),
     [allPermissionTypes]
   )
+  const {bucket} = useContext(WriteDataDetailsContext)
+
+  useEffect(() => {
+    dispatch(getBuckets())
+  }, [])
+
+  useEffect(() => {
+    onSelectBucket(bucket.name)
+  }, [bucket])
```

Comment:
`onSelectBucket` could change based on changing props, so it should go into the dependency array.